crates/api/src/local_user.rs

```diff
@@ -189,6 +194,12 @@ impl Perform for SaveUserSettings {
       }
     }
 
+    if let Some(Some(matrix_user_id)) = &matrix_user_id {
+      if !is_valid_matrix_id(matrix_user_id) {
```

Comment:
Maybe trim() before checking/storing?

Reply:
I moved that to the regex, using `^ ... $`. A unit test makes sure that any one with spaces won't validate.